elemica-suggest.coffee

```diff
@@ -154,10 +160,11 @@ elemicaSuggest 0.7.1-SNAPSHOT.
         $(this).attr 'autocomplete', 'off'
 
         $(this).on 'blur', (event) ->
-          removeSuggestions event.target
+          $target = $(event.target)
+          removeSuggestions $target
 
           if $valueInput.val() == ""
-            $(event.target).val("")
+            $target.val("") if noSuggestionMatched($target.val())
             afterSelect(null)
```

Comment:
No, the contract we've held to is that in the event the event no selection was made and that was final then `afterSelect(null)` is invoked. That gives the user the ability to have one place to respond to the selection that was made. (e.g. taking whatever is passed into after select and setting it on a Knockout observable)

`afterSelect` is also passed `null` in the event the user backspaces.